src/main/resources/assets/tconstruct/book/puny_smelting/sections/upgrades.json

```diff
@@ -119,5 +119,48 @@
   {
     "type": "modifier",
     "data": "upgrades/harvest/magnetic.json"
+  },
+  
+  {
+    "name": "group_armor",
+    "type": "text",
+    "data": "upgrades/armor.json"
+  },
+  {
+    "name": "feather_falling",
+    "type": "modifier",
+    "data": "upgrades/armor/feather_falling.json"
+  },
+  {
+    "type": "modifier",
+    "data": "upgrades/armor/leaping.json"
+  },
+  {
+    "type": "modifier",
+    "data": "upgrades/armor/power.json"
+  },
+  {
+    "type": "modifier",
+    "data": "upgrades/armor/respiration.json"
+  },
+  {
+    "type": "modifier",
+    "data": "upgrades/armor/ricochet.json"
+  },
+  {
+    "type": "modifier",
+    "data": "upgrades/armor/speedy.json"
+  },
+  {
+    "type": "modifier",
+    "data": "upgrades/armor/springy.json"
+  },
+  {
+    "type": "modifier",
+    "data": "upgrades/armor/sticky.json"
```

Comment:
Should also be added to melee. I was lazy in the encyclopedia and just added the same page twice